assets/src/edit-story/app/story/useStoryReducer/test/updateElementById.js

```diff
@@ -39,6 +39,100 @@ describe('updateElementById', () => {
     ]);
   });
 
+  it('should update the given element via an update function', () => {
+    const { restore, updateElementById } = setupReducer();
+
+    // Set an initial state with a current page with an element.
+    restore({
+      pages: [{ id: '111', elements: [{ id: '123' }, { id: '456' }] }],
+      current: '111',
+    });
+
+    const result = updateElementById({
+      elementId: '123',
+      properties: () => ({ a: 1 }),
+    });
+
+    expect(result.pages).toStrictEqual([
+      { id: '111', elements: [{ id: '123', a: 1 }, { id: '456' }] },
+    ]);
+  });
+
+  it('should skip reserved properties when updating by function', () => {
+    const { restore, updateElementById } = setupReducer();
+
+    // Set an initial state with a current page with an element.
+    restore({
+      pages: [{ id: '111', elements: [{ id: '123' }, { id: '456' }] }],
+      current: '111',
+    });
+
+    const result = updateElementById({
+      elementId: '123',
+      properties: () => ({ a: 1, id: '321' }),
+    });
+
+    expect(result.pages).toStrictEqual([
+      { id: '111', elements: [{ id: '123', a: 1 }, { id: '456' }] },
+    ]);
+  });
+
+  it('should do nothing if update by function only attempts reserved attributes', () => {
+    const { restore, updateElementById } = setupReducer();
+
+    // Set an initial state with a current page with an element.
+    const initial = restore({
+      pages: [{ id: '111', elements: [{ id: '123' }, { id: '456' }] }],
+      current: '111',
+    });
+
+    const result = updateElementById({
+      elementId: '123',
+      properties: () => ({ id: '321' }),
+    });
+
+    expect(result).toStrictEqual(initial);
+  });
+
+  it('should not allow updating reserved properties', () => {
+    const { restore, updateElementById } = setupReducer();
+
+    // Set an initial state with a current page with an element.
+    restore({
+      pages: [
+        {
+          id: '111',
+          elements: [
+            { id: '123', isBackground: true, isDefaultBackground: true, x: 1 },
+            { id: '456' },
+          ],
+        },
+      ],
+      current: '111',
+    });
+
+    // only x=2 is allowed to be changed here
```

Comment:
Is there a reason why the default element attributes should be allowed to be changed at all? Should this just be ignored then? Wouldn't it cause an additional history entry otherwise?

Reply:
Actually only one attribute of the background element can be "legally" changed: `backgroundColor` - which is how you change the color/pattern of said background. All other changes should be disallowed or actually turn it into a non-default-background element (and clone the previous element to the default).

This is how dropping a media element directly from library works - it simply merges all the properties from the media element into the background via `updateElementById` (and this is the cause of other problems).

Reply:
I have now changed this, so that dragging an element from media library directly to an existing element will go through `combineElements` rather than `updateElement` which unifies this handling. Please re-review after 4594f129.

Reply:
OK, I was wondering why "drag along" the `defaultBackgroundElement` if it's all static except for the `id` but looks like the background color needs changing, indeed.